### PR TITLE
Add once modifier. Add CSS selector

### PR DESCRIPTION
**Pull request by peterszerzo**

### CODE REVIEW

File: packages/journey-manager/src/index.ts [hunk 1]
```diff
@@ -87,6 +116,22 @@ const matchesUrlCondition = (urlCondition: UrlCondition): boolean => {
...
+  const triggeredSteps = getTriggeredSteps(config.conversationId);
+
+  const sendStep = (stepId: string, once: boolean): void => {
+    if (triggeredSteps.includes(stepId)) {
+      triggeredSteps.push(stepId);
+      saveTriggeredSteps(config.conversationId, triggeredSteps);
+      if (once) {
+        return;
+      }
+    }
```
Comment (jakub-nlx) [inline packages/journey-manager/src/index.ts:128]: Something doesn't seem quite right here. How would a stepId ever get into `triggeredSteps` in the first place? Perhaps a `!` missing in the conditional?

File: packages/journey-manager/src/queries.ts [hunk 2]
```diff
@@ -126,12 +126,27 @@ export async function evaluate(q: Query): Promise<boolean> {
...
 
-  const methodName: keyof typeof queryFns = `findBy${q.queryName}`;
-
-  return await queryFns[methodName](container, ...(q.queryArgs as QueryArgs));
+  let results: HTMLElement[][] = [];
+  if (q.queryName === "QuerySelector") {
+    results = await Promise.all(
+      containers.map((container) => [
+        ...container.querySelectorAll<HTMLElement>(q.queryArgs[0] as string),
+      ]),
```
Comment (jakub-nlx) [inline packages/journey-manager/src/queries.ts:137]:
```suggestion
       containers.map((container) =>
        waitFor(
          () => {
            const els = [
              ...container.querySelectorAll(q.queryArgs[0] as string),
            ];
            if (els.length == 0) {
              throw new Error("No elements found");
            }
          },
          { container },
        ),
      ),
```

where `waitFor` comes from `"@testing-library/dom"`

**Review by jakub-nlx — CHANGES_REQUESTED** [review]

File: packages/journey-manager/src/queries.ts (hunk 2 above)
Comment (peterszerzo) [reply packages/journey-manager/src/queries.ts:137]: @jakub-nlx how do you get away without returning `els`, am I missing something? This is what I could come up with:

```ts
      containers.map((container) =>
        waitFor(
          () => {
            const els = [
              ...container.querySelectorAll<HTMLElement>(
                q.queryArgs[0] as string,
              ),
            ];
            if (els.length == 0) {
              throw new Error("No elements found");
            }
            return els;
          },
          { container },
        ),

```

File: packages/journey-manager/src/index.ts (hunk 1 above)
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts]: You're totally right, updated the logic!

File: packages/journey-manager/src/queries.ts (hunk 2 above)
Comment (jakub-nlx) [reply packages/journey-manager/src/queries.ts]: Oh yeah, you definitely want to return `els` 🤦

File: packages/journey-manager/src/queries.ts [hunk 3]
```diff
@@ -126,12 +127,42 @@ export async function evaluate(q: Query): Promise<boolean> {
...
+            { container },
+          ),
+      ),
+    );
+  } else {
+    const methodName: keyof typeof queryFns = `findAllBy${q.queryName}`;
+    results = await Promise.all(
+      containers.map(
+        async (container) =>
+          // TODO improve typing
```
Comment (jakub-nlx) [inline packages/journey-manager/src/queries.ts:160]:
```suggestion
```

Typing now seems fine, no?

**Review by jakub-nlx — APPROVED** [review]

**Review by jakub-nlx — APPROVED** [review]

**Pull request merged**